makefiles/cflags.inc.mk

```
@@ -39,9 +39,10 @@ endif
 CFLAGS += -fno-common
 
 # Compress debug info. This saves approximately 50% of disk usage.
-# It has no effect if debugging information is not emitted, so it can be left
-# on unconditionally.
-OPTIONAL_CFLAGS += -gz
+# This makes ccache less effective, so disable for CI builds.
+ifneq (1, $(RIOT_CI_BUILD))
```

If `ccache` is the problem and not CI, then the guard should be `ccache` specific, no?
```suggestion
ifeq (, $(CCACHE))
```

---

That seems reasonable, but on my laptop, the performance hit is only ~20%, not 50%. It is quite beefy compared to the multicore CI workers (per-core wise). This might be more affecting build performance on those CPU or memory bound setups.

Also, ccache doesn't compress cache contents by default ("compression=false"), so this PR does save 50% space. (even when no debug symbols are generated...) CI enables ccache compression, which already brings down cache content size to the same level as with "-gz".

So I think locally this is might be useful.

That said, shouldn't using ccache be the first hint to every RIOT developer? Is any of you guys *waiting for local single builds* longer than a couple of seconds??

---

Mh... Then maybe a combination of the two. I did not consider that `ccache` is also run locally sometimes (no, I don't use it, only when doing multi-board builds, because having just a small-ish SSD in my PC here, I value HDD space over time (also I have an octo-core so it is pretty fast anyway) ;-P). I was just thinking that the dependency is not "the CI", but `ccache` in this instance. There are other CIs used in our ecosystem than Murdock and not all use `ccache`. 

Maybe a combination of the two then?

```suggestion
ifneq (1,$(RIOT_CI_BUILD))
ifeq (,$(CCACHE))
```

---

> having just a small-ish SSD in my PC here, I value HDD space over time (also I have an octo-core so it is pretty fast anyway) ;-P).

Does it have a lot of RAM? 4-8G of ccache in tmpfs makes that octocore FLY!

---

Good thinking!

---

I'm in the process of nailing down the problem within the ccache source.
It seems like as soon as any "-g*" option is used, ccache assumes debug info is generated, and includes the current cwd in the hash that identifies cache entries. For "-gz" without any other "-g" other than "-g0", this is a ccache bug. CI builds with a seperate builddir per checkout, so that could be the reason why it is affected a lot more than a local build.

I'm just now trying to create a reproducer and maybe open an upstream bug. For now, I'm confident this is a corner case of how ccache is used in our CI...

Hm, I'm realizing that local builds in different build directories would be affected too. I do that all the time. :) So maybe guarding that by ccache is a good idea.

---

I second @miri64 , the test should not be for RIOT_CI_BUILD. The current meaning of RIOT_CI_BUILD is "turn off any non-determinism". I may want to build with both RIOT_CI_BUILD and debug. Also, the issue may not manifest itself if the path is not changed, which means even testing for CCACHE is too much.

This should be done in a way that is specific to murdock and nothing else.